### DISCUSSION TITLE
Homebrew uses SDKman for JDK & JRE dependencies - such as openjdk?

### DISCUSSION THREAD

**Discussion by theschles** [Homebrew/discussions#5511] [Everyday usage]
### Output of `brew config`

```shell
➜ brew config
HOMEBREW_VERSION: 4.3.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c3ae0327c01aab461531bfc0fcdfb4dd3a779205
Last commit: 2 days ago
Core tap JSON: 20 Jul 18:48 UTC
Core cask tap JSON: 20 Jul 18:48 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.k2GUHZ77xu/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
➜ brew doctor
Your system is ready to brew.
```

### Description of issue

I'm using SDKman for managing my JDK and JRE installations -- and yet Homebrew seems to want to install and manage its own openjdk versions where needed.  In my case:

```
brew uses --installed --recursive java11
pdftk-java
```

How do I configure Homebrew so that -- instead of using Homebrew formulae for JDK / JRE dependencies -- to instead use SDKman for those JDK / JRE dependencies?

**Comment by SMillerDev** [comment]
> How do I configure Homebrew so that -- instead of using Homebrew formulae for JDK / JRE dependencies -- to instead use SDKman for those JDK / JRE dependencies?

There isn't a homebrew way to do this since switching out dependencies for random other software isn't really supported. In the Java world a lot of tools check JAVA_HOME before they check PATH though. So that will help somewhat.
